src/nodemetadata.cpp

```diff
@@ -113,13 +113,13 @@ int NodeMetadata::countNonPrivate() const
 */
 
 void NodeMetadataList::serialize(std::ostream &os, u8 blockver, bool disk,
-	bool absolute_pos) const
+	bool absolute_pos, bool include_empty) const
 {
 	/*
 		Version 0 is a placeholder for "nothing to see here; go away."
 	*/
 
-	u16 count = countNonEmpty();
+	u16 count = include_empty ? m_data.size() : countNonEmpty();
```

Comment:
perhaps `!disk` should automatically turn on `include_empty`.
would that work?

Reply:
It would lead to some unnecessary inclusion of empty metadata. When a mapblock is being serialized to send to clients, it does not need to include its empty metadata. Empty metadata only needs to be included when encoding metadata changes (where empty metadata means a different thing than absent metadata.)